Framework/test/testChecker.cxx

```diff
@@ -0,0 +1,71 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    testChecker.cxx
+/// \author  Piotr Konopka
+///
+
+#include "QualityControl/CheckerFactory.h"
+#include "QualityControl/Checker.h"
+#include <Framework/DataSampling.h>
+#include <Framework/DataSpecUtils.h>
+
+#define BOOST_TEST_MODULE Checker test
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+#include <iostream>
+
+using namespace o2::quality_control::checker;
+using namespace std;
+using namespace o2::framework;
+using namespace o2::header;
+
+BOOST_AUTO_TEST_CASE(test_checker_factory)
+{
+  std::string configFilePath =
+    std::string("json:/") + getenv("QUALITYCONTROL_ROOT") + "/tests/testInfrastructureGenerator.json";
```

Comment:
This test, `testTaskRunner` and `testInfrastructureGenerator` share the same config file, but if needed, I can create separate config files for them. Advantage: maybe clearer project, disadvantage: more files to mantain. What do you think?

Reply:
I suggest to keep a single one for the time being. Just rename it so that we understand it is not only for testInfrastructureGenerator.